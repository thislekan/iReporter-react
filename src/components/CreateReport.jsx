import React from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';
import style from '../styles/CreateReport.css';

function CreateReport(props) {
  const {
    handleChange,
    handleCreateIncident,
    handleMedia,
    disableFileInput,
    clearBlobs,
    imageBlobs,
    // videoBlob,
  } = props;
  return (
    <div id="create-form-div" className={style.createReport}>
      <div className={style.create}>
        <h2>Report an incident</h2>
        <form onSubmit={handleCreateIncident}>
          <div className={style['create-content']}>
            <div className={style['input-div']}>
              <label htmlFor="type">Select type of report</label>
              <select
                name="type"
                id="report-type"
                onBlur={handleChange}
              >
                <option value="">Please select your report type</option>
                <option value="red-flag">Red Flag</option>
                <option value="intervention">Intervention</option>
              </select>
            </div>
            <div className={style['input-div']}>
              <label htmlFor="title">Title</label>
              <input
                type="text"
                name="title"
                id="create-incident-title"
                onBlur={handleChange}
              />
            </div>
            <div className={cx(style['input-div'], style['for-text'])}>
              <label htmlFor="comments">Enter comments</label>
              <textarea
                name="comment"
                id="comments"
                onBlur={handleChange}
              />
            </div>
            <div className={style['input-div']}>
              <label htmlFor="location">Enter Location or Coordinates</label>
              <input
                type="text"
                name="location"
                id="location"
                onBlur={handleChange}
              />
            </div>
            <div className={style['input-div']}>
              <label htmlFor="image or video">Upload Image or Video</label>
              <input
                type="file"
                name="image or video"
                multiple id="file-upload"
                accept="image/*, video/*"
                onClick={clearBlobs}
                onChange={handleMedia}
                disabled={disableFileInput}
              />
            </div>
            <div className={style['media-div']}>
              {/* <div className={style['preview-video']}>
                <video>
                  <track kind="captions" />
                  <source src={videoBlob} />
                </video>
              </div> */}
              <div className={style['preview-img']}>
                {imageBlobs.map((image, index) => <img src={image} key={index} alt="preview" />)}
              </div>
            </div>
            <div className={style['btn-div']}>
              <button
                type="submit"
                id="create-incident"
                className={style['create-incident-btn']}
              >
                Create
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  );
}

CreateReport.propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleCreateIncident: PropTypes.func.isRequired,
  handleMedia: PropTypes.func.isRequired,
  disableFileInput: PropTypes.bool.isRequired,
  clearBlobs: PropTypes.func.isRequired,
  imageBlobs: PropTypes.array.isRequired,
  // videoBlob: PropTypes.string.isRequired,
};

export default CreateReport;
