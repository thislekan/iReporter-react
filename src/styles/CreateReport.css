#create-form-div{
	position: fixed;
	background: white;
	padding: 1rem;
	top: 6rem;
	width: calc(100% - 4rem);
	box-shadow: 0px 0px 2px 0px hsla(245, 8%, 82%, 1);
	z-index: 2;
}
#create-incident{
	background: #117d5a;
	border: 1px solid #117d5a;
	width: 9rem;
	color: white;
	letter-spacing: .5px;
}
.input-div {
	display: flex;
	flex-direction: column;
	padding: 1rem .5rem;
}
.preview-img {
	display: grid;
	grid-template-columns: 1fr 1fr;
	grid-template-rows: auto;
	grid-gap: 1rem;
	padding: .5rem;
	margin-top: 2rem;
}
.preview-img img{
	height: 19.5rem;
	width: 100%;
	object-position: center;
	object-fit: scale-down;
	border: 1px solid gainsboro;
}
.map-region{
	height: 20rem;
	border: 1px solid;
	position: relative;
}
.view-btn{
	background: #23948c;
	border: 1px solid #23948c;
	color: white;
}
.delete-btn{
	border: 1px solid #cf4241;
	background: #cf4c41;
	color: white;
}
.edit-btn{
	background: #117d5a;
	border: 1px solid #117d5a;
	color: white;
}
.comment-div {
    margin: 1rem 0;
}
.btn-div{
	text-align: right;
}
.create-incident-btn{
	background: #117d5a;
	border: 1px solid #117d5a;
	width: 9rem;
	color: white;
	letter-spacing: .5px;
	height: 3rem;
	border-radius: .25rem;
	margin: 1rem 0;
	margin-left: 1rem;
}
select, input, textarea{
	height: 3rem;
	margin: 0.5rem 0;
	border: none;
	background: hsl(210, 9%, 96%);
	padding: .25rem .5rem;
	border-radius: .25rem;
	outline: none;
	color: #117d5a;
	width: 100%;
}
textarea {
	height: 10rem;
	border: 1px solid rgba(29, 57, 74, 0.3);
	font-family: 'Roboto', 'sans-serif', sans-serif;
	letter-spacing: .1rem;
	padding: 2px .5rem;
}

@media only screen and (min-device-width: 700px){
	.control{
		display: grid;
		grid-template-columns: 26rem auto;
		grid-column-gap: 1rem;
		align-items: baseline;
	}
	.image-container img{
		height: 12rem;
	}
	.list-of-reports{
		display: grid;
		grid-template-columns: 1fr 1fr;
		grid-gap: 1rem;
	}
	.map-region{
		height: 30rem;
	}
	.list-of-reports{
		display: grid;
		grid-template-columns: 1fr 1fr;
		grid-gap: 1rem;
	}
	.select-report select{
		width: 25rem;
	}
	#detailed-incident{
		width: calc(100% - 45%);
	}
	.notification__body{
		width: calc(100% - 40rem);
		left: 20rem;
	}
	#create-form-div{
		top: 7rem;
		width: calc(100% - 40rem);
	}
	.create{
		height: 80vh;
	}
	/* .delete-prompt{
			width: calc(100% - 45rem);
			top: calc(50vh - 7.5rem);
	} */
	#filter-div{
		width: unset;
	}
}
@media only screen and (min-device-width: 900px){
	.padded-body{
		padding: 5rem 10rem;
	}
	.notification__body{
		width: calc(100% - 70rem);
		left: 35rem;
	}
	.image-container img{
		height: 22rem;
	}
	.overlay{
		width: calc(100% - 20rem);
	}
	/* .delete-prompt {
			width: calc(100% - 70rem);
	} */
	.create{
		height: 82vh;
	}
	#create-form-div{
		width: calc(100% - 70rem);
	}
}