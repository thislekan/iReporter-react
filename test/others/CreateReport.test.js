import React from 'react';
import { shallow } from 'enzyme';
import CreateReport from '../../src/components/CreateReport.jsx';

const mockFn = jest.fn();
describe('<CreateReport />', () => {
  it('should render the index', () => {
    const wrapper = shallow(
      <CreateReport
        handleChange={mockFn}
        handleCreateIncident={mockFn}
        handleMedia={mockFn}
        disableFileInput
        clearBlobs={mockFn}
        imageBlobs={['jd', 'djh']}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
