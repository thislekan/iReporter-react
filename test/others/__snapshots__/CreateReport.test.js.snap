// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateReport /> should render the index 1`] = `
<div
  id="create-form-div"
>
  <div>
    <h2>
      Report an incident
    </h2>
    <form
      onSubmit={[MockFunction]}
    >
      <div>
        <div>
          <label
            htmlFor="type"
          >
            Select type of report
          </label>
          <select
            id="report-type"
            name="type"
            onBlur={[MockFunction]}
          >
            <option
              value=""
            >
              Please select your report type
            </option>
            <option
              value="red-flag"
            >
              Red Flag
            </option>
            <option
              value="intervention"
            >
              Intervention
            </option>
          </select>
        </div>
        <div>
          <label
            htmlFor="title"
          >
            Title
          </label>
          <input
            id="create-incident-title"
            name="title"
            onBlur={[MockFunction]}
            type="text"
          />
        </div>
        <div
          className=""
        >
          <label
            htmlFor="comments"
          >
            Enter comments
          </label>
          <textarea
            id="comments"
            name="comments"
            onBlur={[MockFunction]}
          />
        </div>
        <div>
          <label
            htmlFor="location"
          >
            Enter Location or Coordinates
          </label>
          <input
            id="location"
            name="location"
            onBlur={[MockFunction]}
            type="text"
          />
        </div>
        <div>
          <label
            htmlFor="image or video"
          >
            Upload Image or Video
          </label>
          <input
            accept="image/*, video/*"
            disabled={true}
            id="file-upload"
            multiple={true}
            name="image or video"
            onChange={[MockFunction]}
            onClick={[MockFunction]}
            type="file"
          />
        </div>
        <div>
          <div>
            <img
              alt="preview"
              key="0"
              src="jd"
            />
            <img
              alt="preview"
              key="1"
              src="djh"
            />
          </div>
        </div>
        <div>
          <button
            id="create-incident"
            type="submit"
          >
            Create
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
